src/components/Calculator.js

```diff
@@ -1,67 +1,72 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
+import calculate from '../logic/calculate';
 import './Calculator.css';
 
-function Op({ operation }) {
+function Op({ operation, cb }) {
   return (
-    <button type="button" className="operation">
+    <button type="button" className="operation" onClick={cb}>
       {operation}
     </button>
   );
 }
 
-function Number({ num }) {
+function Number({ num, cb }) {
   const style = {};
   if (num === '0') {
     style['grid-column'] = 'span 2';
   }
 
   return (
-    <button type="button" style={style} className="number">
+    <button type="button" style={style} className="number" onClick={cb}>
       {num}
     </button>
   );
 }
 
 Number.propTypes = {
   num: PropTypes.string.isRequired,
+  cb: PropTypes.func.isRequired,
 };
 
 Op.propTypes = {
   operation: PropTypes.string.isRequired,
+  cb: PropTypes.func.isRequired,
 };
 
-const numberSectionButtons = [];
-['AC', '+/-', '%', 7, 8, 9, 4, 5, 6, 1, 2, 3, 0, '.'].forEach((el) => {
-  numberSectionButtons.push(<Number num={`${el}`} />);
-});
+function Calculator() {
+  const [state, setState] = useState({
+    total: 0,
+    next: 0,
+    operation: '',
+  });
+  const numberSectionButtons = [];
+  ['AC', '+/-', '%', 7, 8, 9, 4, 5, 6, 1, 2, 3, 0, '.'].forEach((el) => {
+    numberSectionButtons.push(<Number
+      num={`${el}`}
+      cb={() => {
+        setState((old) => calculate(old, `${el}`));
+      }}
+    />);
+  });
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
-  }
-
-  render() {
-    return (
-      <div className="calculator">
-        <input className="quote" placeholder="0" />
-        <div className="buttons">
-          <div className="number-section">
-            {numberSectionButtons}
-          </div>
-          <div className="operation-section">
-            <Op operation="÷" />
-            <Op operation="X" />
-            <Op operation="-" />
-            <Op operation="+" />
-            <Op operation="=" />
-          </div>
+  return (
+    <div className="calculator">
+      <input className="quote" placeholder="0" value={state.next ? state.next : state.total} readOnly />
+      <div className="buttons">
+        <div className="number-section">
+          {numberSectionButtons}
+        </div>
+        <div className="operation-section">
+          <Op operation="÷" cb={() => setState((old) => ({ ...old, ...calculate(old, '÷') }))} />
+          <Op operation="x" cb={() => setState((old) => ({ ...old, ...calculate(old, 'x') }))} />
+          <Op operation="-" cb={() => setState((old) => ({ ...old, ...calculate(old, '-') }))} />
+          <Op operation="+" cb={() => setState((old) => ({ ...old, ...calculate(old, '+') }))} />
+          <Op operation="=" cb={() => setState((old) => ({ ...old, ...calculate(old, '=') }))} />
```

Comment:
[OPTIONAL] I think that instead of defining an anonymous function in each component we could extract the code that it's common between functions and pass just a single function. This would result in a more DRY code. [Read more about this important concept here.](https://qvault.io/clean-code/dry-code/)